src/main/java/hudson/plugins/s3/S3BucketPublisher.java

```diff
@@ -236,9 +259,9 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launc
                 run.addAction(new S3ArtifactsAction(run, profile, artifacts));
                 run.addAction(new FingerprintAction(run, record));
             }
-        } catch (IOException e) {
+        } catch (Exception e) { // Any failure in this plugin should be constrained according to config
```

Comment:
Why do you need to suppress `InterruptedException`? It could break a lot of stuff. Because it would continue execution then it's not desired.
If build was interrupted by Jenkins (i.e. aborted from timeout plugin, executor was stopped from UI, etc.), it doesn't have chance to pass (at least I can't imagine such case).